tools/bundletool/process_and_sign.sh.template

```diff
@@ -30,6 +30,40 @@ if [[ -n "$POST_PROCESSOR" ]]; then
   "$POST_PROCESSOR" "$WORK_DIR"
 fi
 
+# Strip out any unnecessary slices from embedded dynamic frameworks to save space
+for i in "$WORK_DIR"/*.app; do
+    # Lets figure out what slices the app has
+    app_name=$(basename $i .app)
+    slices=$(xcrun lipo -info $i/$app_name | cut -d: -f3)
+
+    # Now lets make sure any included frameworks match
+    for f in $i/Frameworks/*.framework; do
+        # We need to check we have overlapping architectures
+        framework_name=$(basename $f .framework)
+        framework_slices=$(xcrun lipo -info $f/$framework_name | cut -d: -f3)
+        # Sanity check that the framework has at least the slices the app has
+        for s in $slices ; do
```

Comment:
Should this script be responsible for this, or should this fall through to be a linker error?

Reply:
Or wait, this happens after linking, right? Does this even need to be checked then?

Reply:
Hmm, that is a good point, the linker will fail if there isn't enough slices. This change was really around removing unnecessary slices, I just figured it didn't hurt to be sure.... but you are right, it can't happen without some serious issues up stream.